analyzer/tests/functional/ctu_failure/test_ctu_failure.py

```diff
@@ -229,16 +360,48 @@ def test_ctu_fallback(self):
         # Ctu should fail during analysis of main.c
         self.assertIn("main.c", failed_files[0])
 
-    def __do_ctu_all(self, reparse, extra_args=None):
+    def test_ctu_on_demand_fallback(self):
+        """ In case of ctu failure the non ctu analysis will be triggered when
+        using on-demand-mode.
+        """
+        self.__set_up_test_dir('ctu_failure')
+        if not self.ctu_capable:
+            self.skipTest(NO_CTU_MESSAGE)
```

Comment:
Cannot we use the `@unittest.skip` decorator to skip this test case based on the given conditions?

For more info see: https://docs.python.org/3/library/unittest.html

Reply:
Decorators rule, thanks for pointing them out. I have implemented the test skipping that way.